src/main/java/com/starrocks/connector/flink/manager/StarRocksQueryPlanVisitor.java

```diff
@@ -112,6 +113,8 @@ private static QueryPlan getQueryPlan(String querySQL, String httpNode, StarRock
                 HttpPost post = new HttpPost(url);
                 post.setHeader("Content-Type", "application/json;charset=UTF-8");
                 post.setHeader("Authorization", getBasicAuthHeader(sourceOptions.getUsername(), sourceOptions.getPassword()));
+                Optional.ofNullable(sourceOptions.getWarehouseName())
+                        .ifPresent(warehouse -> post.setHeader("warehouse", warehouse));
```

Comment:
@kevincai I refine this code to avoid setting warehouse header(null value) in case of flink sink tale not having `'scan.params.warehouse'='warehosue_name'`.  And then fe will use the `default_warehouse` if not getting the warehouse header.